### DISCUSSION TITLE
recurring changed with delay trigger

### DISCUSSION THREAD

**Discussion by jimtng** [boc-tothefuture/openhab-jruby#138] [Ideas]
It would be handy to have:

```ruby
rule 'x' do
  changed Item1, to: ON, for: 10.minutes, repeat: N
  run { logger.info ("Item1 is ON for a while") }
end
```

Where: N: default = 1 it would run the rule once
N = 2: it would run the rule twice, at minute 10 and minute 20 (as long as Item1 stays on).
N = -1 (or infinity?): It would run the rule every 10 minutes for as long as Item1 stays on.

**Comment by boc-tothefuture** [comment]
This is interesting..
Can you provide a real world use case?

**Reply by jimtng** [reply]
Case:

I want to notify via say "Warning, such and such doors are open", and I want to keep sending this notification every X minutes until the doors are closed. So it needs a repeating timer.

**Reply by boc-tothefuture** [reply]
I am in on the concept.

A couple of questions..   Does the `for` duration need to match the repeat interval.  I could think of some scenarios (in that specific door example) where you don't want it to.  You might want to be reminded every hour or every minute depending on your use case.  

In that case would we want to extend every syntax?

```
rule 'x' do
  changed Item1, to: ON, for: 10.minutes, every: 5.minutes
  run { logger.info ("Item1 is ON for a while") }
end
```

**Reply by jimtng** [reply]
> In that case would we want to extend every syntax?
> 
> ```
> rule 'x' do
>   changed Item1, to: ON, for: 10.minutes, every: 5.minutes
>   run { logger.info ("Item1 is ON for a while") }
> end
> ```
> 
> 

This looks good. Even better than my original suggestion

**Reply by jimtng** [reply]
How would we handle this case when `every:` is given but `for:` is omitted:
```ruby
rule ' x' do
  changed Item1, to: ON, every: 5.minutes
  run { ... }
end
```

Would this translate to for = every? i.e. would it be the same as:
```ruby
rule ' x' do
  changed Item1, to: ON, for: 5.minutes, every: 5.minutes
  run { ... }
end
```

When you read it, the first rule above seems to mean that Item1 is changed to ON every 5 minutes, i.e. "flapping" rather than what the second rule would mean.
